### DISCUSSION TITLE
about quota

### DISCUSSION THREAD

**Discussion by tuanchuoi** [versity/versitygw#762] [Q&A]
hi guys,
Can we set a quota for users, such as less than 100 GB?

**Comment by benmcclelland** [comment]
The design of the gateway is such that there can be multiple gateways hosting the same filesystem data. This means requests can be load balanced across multiple gateways. The drawback here is that we don't really have a way to track what all data was written by any user from the point of view of any individual gateway.

Ay quota management would most likely be the responsibility of the backend system. However, we typically run the gateway as root and root access may defeat any quota enforcement for some filesystems.

We have [options](https://github.com/versity/versitygw/wiki/POSIX-Backend#args) to set uid/gid of a file to the request access account uid/gid. But I think this gets set after upload. We may want to think about setting the ownership on create, and some filesystems might use the file ownership to enforce quotas.
